seaborn/_oldcore.py

```diff
@@ -922,7 +924,7 @@ def _assign_variables_longform(self, data=None, **kwargs):
                     val in data
                     or (isinstance(val, (str, bytes)) and val in index)
                 )
-            except (KeyError, TypeError):
+            except (KeyError, TypeError, ValueError):
```

Comment:
if `val` is a non-pandas Series, then checking `val in data` will throw `ValueError`